src/solver/hydro/management/management.h

```diff
@@ -42,39 +42,84 @@ class State;
 }
 } // namespace Solver
 
+enum
+{
+    //! The maximum number of days in a year
+    dayYearCount = 366
+};
+
+//! Temporary data
+struct TmpDataByArea
+{
+    //! Monthly local effective demand
+    double MLE[12];
+    //! Monthly optimal generation
+    double MOG[12];
+    //! Monthly optimal level
+    double MOL[12];
+    //! Monthly target generations
+    double MTG[12];
+    //! inflows
+    double inflows[12];
+    //! monthly minimal generation
+    std::array<double, 12> mingens;
+
+    //! Net demand, for each day of the year, for each area
+    double DLN[dayYearCount];
+    //! Daily local effective load
+    double DLE[dayYearCount];
+    //! Daily optimized Generation
+    double DOG[dayYearCount];
+    //! daily minimal generation
+    std::array<double, dayYearCount> dailyMinGen;
+
+    // Data for minGen<->inflows preChecks
+    //! monthly total mingen
+    std::array<double, 12> totalMonthMingen;
+    //! monthly total inflows
+    std::array<double, 12> totalMonthInflows;
+    //! yearly total mingen
+    double totalYearMingen;
+    //! yearly total inflows
+    double totalYearInflows;
+
+}; // struct TmpDataByArea
+
+typedef struct
+{
+    std::vector<double> HydrauliqueModulableQuotidien; /* indice par jour */
+    std::vector<double> NiveauxReservoirsDebutJours;   //Niveaux (quotidiens) du reservoir de d�but
+    //de jour (en cas de gestion des reservoirs).
+    std::vector<double> NiveauxReservoirsFinJours; //Niveaux (quotidiens) du reservoir de fin
+    //de jour (en cas de gestion des reservoirs).
+} VENTILATION_HYDRO_RESULTS_BY_AREA;
+
+// vector of [numSpace][area]
+using ALL_HYDRO_VENTILATION_RESULTS = std::vector<std::vector<VENTILATION_HYDRO_RESULTS_BY_AREA>>;
+
+
 class HydroManagement final
 {
 public:
-    //! \name Constructor & Destructor
-    //@{
-    /*!
-    ** \brief Default constructor
-    */
-    explicit HydroManagement(Data::Study& study);
-    //! Destructor
+    HydroManagement(const Data::AreaList& areas,
+                    const Data::Parameters& params,
+                    const Date::Calendar& calendar,
+                    unsigned int maxNbYearsInParallel,
+                    Solver::IResultWriter::Ptr resultWriter);
+
     ~HydroManagement();
-    //@}
 
-    //! Get an initial reservoir level
     double randomReservoirLevel(double min, double avg, double max);
 
     //! Perform the hydro ventilation
-    void operator()(double* randomReservoirLevel,
-                    Solver::Variable::State& state,
-                    uint y,
-                    uint numSpace,
-                    VAL_GEN_PAR_PAYS& valeursGenereesParPays);
-
-public:
-    //! Random number generator
-    MersenneTwister random;
+    void makeVentilation(double* randomReservoirLevel,
+                        Solver::Variable::State& state,
+                        uint y,
+                        uint numSpace);
```

Comment:
```suggestion
    ALL_HYDRO_VENTILATION_RESULTS makeVentilation(double* randomReservoirLevel,
                        Solver::Variable::State& state,
                        uint y,
                        uint numSpace);
```
One single method instead of two.

Reply:
ok. So this means that we could simplify the caller too : 
```C++
// solver.hxx
yearFailed[y] = !simulation_->year(...,
                                   ...,
                                   simulation_->hydroManagement.ventilationResults());
```

Reply:
I think we should keep the timer, but have a single function.
```cpp
// Timer ON
auto hydroVentilation = simulation_->hydroManagement.makeVentilation(/* args */);
// Timer OFF
yearFailed[y] = !simulation_->year(...,
                                   ...,
                                   hydroVentilation);
```